### DISCUSSION TITLE
陽性登録情報の可視化：陽性登録タイムラグの推定

### DISCUSSION THREAD

**Discussion by parasite2006** [cocoa-mhlw/cocoa#831] [General]
以下は接触確認アプリ本体のコードとは直接関係ないものの、アプリの効果を考える上で参考としてご覧いただければと考えて投稿させていただきました。

私は「陽性登録者の発症/検査日から登録実行までどのくらいのタイムラグがあるのか、その結果陽性登録者と接触してから接触通知が届くまでにどのくらいかかっているのか推定する手がかりにならないか」と考えて、@kvaluation さんがGoogle  Data Studio
https://datastudio.google.com/reporting/069598a2-3f01-4b51-b023-cdb478992182/page/blffB
で公開して下さっている推定接触日つきの接触キー数データを1月中旬からデータ配信日ごとにグラフ化するようになりました。

接触確認アプリの接触通知の送信対象時期の範囲と陽性登録用処理番号の請求方法は以下のように変遷しています。
A) 陽性登録日から過去2週間/保健所からの検査結果連絡時に口頭請求
（〜2020年12月14日まで）
B) 発症/検査日の2日前から陽性登録日まで/口頭請求
（2020年12月15日〜2021年6月末）
C) 発症/検査日の2日前から陽性登録日まで/口頭請求+MY HER-SYS IDによるスマホ請求
（2021年7月1日〜2021年11月末）
D) 発症/検査日の2日前から陽性登録日まで/口頭請求+スマホ請求+SMS通知
（2021年12月〜現在）

配信日2020年8月21日（時期A）、2021年8月25日（時期C、2021年夏第5波の最大値）、2022年1月27日（時期D）の3枚のグラフをご覧ください。
<img width="680" alt="推定接触日別接触キー数(20200821)" src="https://user-images.githubusercontent.com/84975567/151439538-632fdd44-9694-4ae6-95ed-99a8e442f16b.png">
<img width="681" alt="推定接触日別接触キー数(20210825)" src="https://user-images.githubusercontent.com/84975567/151439626-f0ed07f5-05ec-4465-a151-971ac51173d7.png">
<img width="682" alt="推定接触日別接触キー数(20220127)" src="https://user-images.githubusercontent.com/84975567/151439682-14b30656-455b-4cb9-acaa-37f42106e7e6.png">

時系列順に見ると、
接触通知の送信対象時期が絞られる前：配信日2日前（推定陽性登録日の前日）から過去に遡るにつれて徐々に減少
接触通知の送信対象時期が絞られた後：配信日2日前から4日前（推定陽性登録日の前日から3日前）の3日間が最も高く、そこから遡ると徐々に減少
処理番号のSMS通知導入後：配信日2日前（推定陽性登録日の前日）が飛び抜けて高い。そこから遡るにつれて減少するが、推定陽性登録日から1週間以上経ってから提供されているキーがまだある。（こういうタイムラグの長いキーは提供されたキー全体に占める割合は低いものの、Twitterのつぶやきには上がってくるので目立つ。タイムラグの長い接触通知は受信者の接触確認アプリに対する印象を確実に悪くし、信頼を損ねている）

接触通知の送信対象時期が絞られている現在、陽性登録者との接触から接触通知受信までの間のどの段階でどの程度のタイムラグが生じうるかを概念図にまとめました。
![PositiveRegistrationTimeLag2](https://user-images.githubusercontent.com/84975567/151440239-cbc86169-a500-42da-a268-5eb20c54c28b.png)

発生届が医療機関でHER-SYS登録されず、FAX送信されて保健所で代行入力された場合の上積みタイムラグ3日間は、2021年7月28日開催の厚労省アドバイザリーボードの西浦先生資料
https://www.mhlw.go.jp/content/10900000/000812898.pdf
p.84の2021年7月の値を採用しました。このタイムラグは感染拡大時にはさらに延長されると考えられます。
<img width="875" alt="HER-SYStimelag" src="https://user-images.githubusercontent.com/84975567/151440447-904620b9-bfea-4c39-9fc3-0dfca5d6eef4.png">
検査から結果判明までのタイムラグもまた、感染拡大時にはさらに延長されることが容易に想像できます。

陽性登録用処理番号のSMS通知導入は陽性登録率の向上には大きく貢献したものの、陽性登録のタイムラグの短縮にはまた別口の方策が必要だと言えます。

余談ながら、英国の接触確認アプリは構想段階の数理モデル研究
https://togetter.com/li/1564332
https://www.science.org/doi/10.1126/science.abb6936
で「新型コロナウイルス感染者の発症から隔離（+接触者の検疫）までの時間間隔が3日まで延びると感染の抑え込み（実効再生産数<1の実現）に失敗する」との結論を得て、陽性登録時の本人確認の厳密性よりも陽性結果判明から登録実行までのタイムラグの短縮を優先する決断を下し、陽性登録プロセスから本人確認段階を排除しています。
https://twitter.com/tyonarock/status/1431019847173873665

**Comment by daisuke-nogami** [comment]
ありがとうございます。
取り急ぎ一点注記すると、2020年8月の時点では、感染可能期間に絞らずTEKを送信していましたので、単純な比較はできないかと思います（ただ、それでもTEK数が減るのは、14日のTEK保存期間を越えていたということで、非常に登録が遅かったことを示していますが…）。

**Comment by kvaluation** [comment]
@parasite2006 さんありがとうございます。
@daisuke-nogami COCOA2を視野に、どんな統計データがあると良いか、情報共有していけたらと思います。率直に、欲しいデータとかお知らせいただけたら、みんなで考えていけるかと思います。

例えば、
Exposure Notification Privacy-preserving Analytics (ENPA) White Paper
https://covid19-static.cdn-apple.com/applications/covid19/current/static/contact-tracing/pdf/ENPA_White_Paper.pdf
にある仕組みは、日本は採用しない決定済みなのか、採用可能性があるのか、知りたいです。

**Reply by keiji** [reply]
いただいたホワイトペーパーは、プライバシーを守ったまま情報を取得する仕組みと理解しています。ENSのような国境をまたいだデータ取得になるとこういった仕組みが必要になるのかもしれませんね。

個人的に（あくまで個人的には）、どのような情報を送るかを明確にした上で、利用者の明確な合意を得て情報を提供してもらう方法でいいのではないかと。むしろ、全部オープンデータにしてしまえばいいのではないかと言っているのですが、それ以前の課題がまだまだあって、そこまでの検討に至っていないと言うのが実際です。

**Reply by daisuke-nogami** [reply]
必要性を強く伝えるためには、どのような分析・施策に繋げられるかを説明する必要があります。
単純にこういうホワイトペーパーがあるから、ではそもそも判断が難しい点はご了承ください。

一方、現状で取得しているデータを元にした統計を使い切っている分けでもないので、
その点からまず掘るべきかと思います。

**Reply by kvaluation** [reply]
ありがとうございます。
■いま、陽性登録者数や、提供されたキーの数は分かりますが、その陽性登録で何人ぐらいに通知がでたのか、その数（接触通知数ととりあえずいいます）の推移がわからないです。
接触通知数がわかると、比率として、何件電話がかかりそうかとか、検査態勢との関係でどうかとか、ご判断頂く際の参考データの１つになりそうに思います。

■検査や診断のキャパシティーとの関係で、感染者数の増加率が見かけ上減っていく（簡易版の実行再生産数が低下する）現象に対して、検査陽性率や、無作為検査(東京都の戦略的検査強化事業・交通結節点)の陽性率や、発熱相談センターの相談件数の推移等で補正するような知見があります。さらに、COCOAの陽性登録遅延などを参酌できれば、各種サンプリングデータから実数・実態を見極めて頂く際に、ご活用いただけるのではないかと想像します。

厚生労働省殿やアドバイザリーボードの先生方に、どのようなデータをお示しできるか、COCOAの動作信頼性も高まってきておりますので、低コストで貢献できることがないか、模索できたらと思います。

**Reply by kvaluation** [reply]
> 全部オープンデータ

賛成です。COCOA2のcocoa-logなどに、DailySummaryの内容などが入るのか、気になっています。

**Reply by b-wind** [reply]
以前にも主張していますが、個人的には役にたつデータはどんどん取得すれば良いとの主張です。
現状では通知の出た数すら取っていないと認識しています。


ただし、これも繰り返しになりますが対象は健康・医療に関わるプライバシーに非常に深く関わるデータです。
とりあえず集めてみてどんな分析が出来るか考える。と言った方針には馴染まないと考えます。


集めたデータでどのような分析が出来、提供側・利用者にとって何のメリット・デメリットがあるか明確にした上での取得が前提になるのかなと思います。
（前例が無いので難しい事とは思いますが）

**Reply by b-wind** [reply]
@daisuke-nogami 

> 一方、現状で取得しているデータを元にした統計を使い切っている分けでもないので、
> その点からまず掘るべきかと思います。

現時点で候補に挙がりそうな物としては、どのような物があるでしょうか？ 例示戴ける物があれば。

**Reply by daisuke-nogami** [reply]
> その陽性登録で何人ぐらいに通知がでたのか、その数（接触通知数ととりあえずいいます）の推移がわからないです。

この件（通知発生件数）については、

- 必要性はすでに #532 でも議論し、実装案は https://github.com/cocoa-mhlw/cocoa/issues/597 で準備いただいている
- 実装の可否については、行政/Apple/Googleの三者の確認が必要である
  - 行政側ではプライバシーポリシー/利用規約に係る議論を担当している会議体にて議論する準備をすすめている
  - Apple/Googleとのやり取りについては現時点でお話しできる情報はない

ということで、進捗がゼロということはないです、とだけお示しします。
（心の中の声：調整が終わるまでは深堀りしようがないので、いま取り得る数字の話を優先してしたい）

**Reply by daisuke-nogami** [reply]
> 現時点で候補に挙がりそうな物としては、どのような物があるでしょうか？

使い切っている、という表現がよろしくなかったですね。
新しい知見を絞り出すというよりは、今回の「陽性登録タイムラグの推定」のような統計情報をちゃんと定常的に可視化（して、他の分析に繋げる）ことにも意味があるのでは、という考えでした。

陽性登録のタイムラグは、登録のタイムラグという部分に着目するとCOCOAに閉じた話ですが、直近HER-SYS登録とCOCOA登録の間のタイムラグが非常に短くなっている現状を踏まえると、「発症/検査結果判明」から「HER-SYS登録=新規感染者数としての認知」のタイムラグと近似することもできます。

現時点でHER-SYS登録などと発症/検査のタイムラグは、そこまでリアルタイムで追えている/発表されているわけではないので、検査結果発表の遅れを加味したよりリアルタイム性高い分析には使えるのではないか、という話です。
（ただ、そのあたりをTEKの束などから統計数値化する仕組みまでは組めていません。適切なログが仕込まれてDBのテーブルが組まれていたら可視化ダッシュボードづくりは私の本職に近い話でいろいろやりうるのですが…）

※というわけで、COCOA運営側としては14日で破棄しているTEKの束を @kvaluation さんが個人的に収集されているのは過去に遡った統計を取るためには重要ではあるのですが…（TEKを登録日・発症/検査日情報だけにして保管したい）

**Reply by b-wind** [reply]
なるほど。既知の情報からも見方によって得られる情報はまだある（はず）と受け取りました。
ご説明ありがとうございます。

**Reply by kvaluation** [reply]
■通知発生件数

@daisuke-nogami 整理をありがとうございます。

★2022年1月末時点
> > その陽性登録で何人ぐらいに通知がでたのか、その数（接触通知数ととりあえずいいます）の推移がわからないです。
> 
> この件（通知発生件数）については、
> 
> * 必要性はすでに [EN v2を搭載するCOCOAで実現したい機能について #532](https://github.com/cocoa-mhlw/cocoa/discussions/532) でも議論し、実装案は [[Server] 接触状況収集APIを試験的に実装する #597](https://github.com/cocoa-mhlw/cocoa/issues/597) で準備いただいている

★2021年4月公開　Exposure Notification Privacy-preserving Analytics (ENPA) White Paper

1 Introductionの第3段落
`For example, ENPA can provide information on how many exposure notification alerts are displayed to users, without exposing how many alerts each device has displayed. `

**仮訳：例えば、ENPA（2021年4月に提案された分析システム）は、各ユーザーのスマホで何回接触通知が表示されたかの情報を隠しつつ、（公衆衛生当局に限定で）通知発生件数を提供できる。**

とあり、この分析からのオプトアウトもOSの設定側に用意されるので、このGoogle | Appleの分析システムを使うことも考えられるかと思います。

**Reply by kvaluation** [reply]
> ※というわけで、COCOA運営側としては14日で破棄しているTEKの束を @kvaluation さんが個人的に収集されているのは過去に遡った統計を取るためには重要ではあるのですが…（TEKを登録日・発症/検査日情報だけにして保管したい）

@daisuke-nogami  2020年9月1日以降、こんなデータがありますので、過去分は、ご指定のライセンス形式でいつでもご提供します。 
必要な際に、covid@kval.jp でもTwitterDMでもご連絡ください。
<img width="1088" alt="スクリーンショット 2022-01-30 13 39 05" src="https://user-images.githubusercontent.com/60703087/151686927-e65650a8-849f-4818-bdc4-ce0604122a72.png">

EN1だと、TEKの発症/検査日はTEKからは読み取れないのかなと考えてました。

**Reply by daisuke-nogami** [reply]
ご確認いただきありがとうございます。

> EN1だと、TEKの発症/検査日はTEKからは読み取れないのかなと考えてました。

はい。そのとおりです。そしてEN2に変えた時点でサーバ側のTEKの様式が変わる点は
https://github.com/cocoa-mhlw/cocoa/issues/792
すでにご覧頂けていますが、それを踏まえた集計の連続性を確保するにはTEKの中身の確認が必要ですね、というご案内でした…

**Comment by b-wind** [comment]
> 発生届が医療機関でHER-SYS登録されず、FAX送信されて保健所で代行入力された場合の上積みタイムラグ3日間

結構厳しいな。と言う印象は受けますね。
HER-SYS の処理性能の問題なのか？医療機関で入力出来ず保健所で代行するも（結局HER-SYSなので）入力出来ないと言う声も漏れ聞こえてきていますし。

**Reply by parasite2006** [reply]
先日HER-SYSにデータ入力が殺到してサーバーダウンした件
（読売新聞によれば1/19夕方と1/20午後の2回
https://www.yomiuri.co.jp/national/20220120-OYT1T50253/
日経新聞によれば1/20
https://www.nikkei.com/article/DGXZQOUA20B9V0Q2A120C2000000/ ）
はシステム側の処理能力の問題で、こちらは一応増強対応が行われているそうです。

これとは別に発生届（様式は全国共通）
https://www.mhlw.go.jp/content/000857749.pdf
の情報量が多い上に、HER-SYSシステムのデータ連携仕様書（これがあれば電子カルテメーカーは、自社製品の電子カルテから発生届の「12.診断方法」の項を書くのに必要な検査結果の記録がある患者のデータを拾って自動的にHER-SYSデータ登録用フォーマットに変換する追加機能を開発できる）の公開も行われていないため、医療機関は電子カルテから情報を肉眼で拾ってHER-SYSデータ登録用画面に手入力する他なく、この手間を嫌って発生届を紙に書いてFAXで保健所に送るのをやめないわけです。
（個人レベルではキーボードショートカット作成ソフトを使って電子カルテの情報をHER-SYSの登録画面にコピペする操作を効率化した例
https://twitter.com/PureChestnutBro/status/1486454669497237504
https://twitter.com/PureChestnutBro/status/1486464692868775937
も見受けられますが、一般に普及しているとは到底言い難い）

兵庫県は県庁内に保健所の事務処理（発生届の代行入力を含む）を遠隔で代行支援する部署を設けたそうですが
https://www.kobe-np.co.jp/news/sougou/202201/0015016201.shtml
どうしても医療機関が発生届のFAX送信を止めないなら、県庁所在地と政令指定都市単位でこういう対応をせざるを得ないでしょう（そしてここにはスキャナーとOCRで紙の発生届をHER-SYS登録データに自動変換するシステムを設置）。

**Comment by keiji** [comment]
@parasite2006 
v2.1.0から陽性情報登録のAPIがV3DiagnosisApiに切り替わる予定です。このAPIを通じて登録された診断キーは、`DaysSinceOnsetOfSymptoms`というパラメーターを持ちます。これはそのキーが症状が現れた日（または診断日）から何日前・後なのかを示す値です（-14, 0, +14の範囲）。

この値を使えば、症状が出てから実際に陽性登録に至るまでの時間を高い精度で推定できるかと思います。

個人的にGitHub Actionsを使って運用している読み込みプログラムを置いておきます（Pythonです）。
https://github.com/keiji/cocoa-statistics/tree/main/opendata-converters/cocoa_diagnosis_keys
